ci/build_cpp.sh

```diff
@@ -17,9 +15,24 @@ rapids-logger "Begin cpp build"
 
 sccache --zero-stats
 
-RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rapids-conda-retry build \
-  conda/recipes/libcuml
+RAPIDS_PACKAGE_VERSION=$(rapids-generate-version)
+export RAPIDS_PACKAGE_VERSION
+
+source rapids-rattler-channel-string
+
+# --no-build-id allows for caching with `sccache`
+# more info is available at
+# https://rattler.build/latest/tips_and_tricks/#using-sccache-or-ccache-with-rattler-build
+rattler-build build --recipe conda/recipes/libcuml \
+                    --experimental \
```

Comment:
Sorry, haven't reviewed one of these PRs before... why are we opting into experimental stuff from `rattler-build`? What specifically do we need?

Reply:
The main usage of `--experimental` is the pre-build or cache-build option, otherwise we'd have to compile everything for each output.  It's been marked experimental for a while, but we're confident it's not going anywhere (I suspect the main reason it's still gated is that upstream may want to change some names around).

Wolf and I were discussing how the phrase [multiple output cache](https://rattler.build/latest/multiple_output_cache/) can cause confusion when paired with `sccache` and compilation caches -- so maybe `pre-build` would be a better name, but that will require breaking changes to the yaml spec.